week-04-associated-data-and-users/day-01/module-01/readme.md

```diff
@@ -0,0 +1,324 @@
+
+```js
+// List model
+
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema,
+    // require Todo model
+    Todo = require('./todo');
+
+var ListSchema = new Schema({
+  name: String,
+  // embed todos in list
+  todos: [Todo.schema]
+});
+
+var List = mongoose.model('List', ListSchema);
+module.exports = List;
+```
+
+```js
+// Todo model
+
+var mongoose = require('mongoose'),
+    Schema = mongoose.Schema;
+
+var TodoSchema = new Schema({
+  text: String,
+  completed: Boolean
+});
+
+var Todo = mongoose.model('Todo', TodoSchema);
+module.exports = Todo;
+```
+
+### Route to Create Embedded Data
+
+```js
+// create todo embedded in list
+app.post('/api/lists/:listId/todos', function (req, res) {
+  // set the value of the list id
+  var listId = req.params.listId;
+
+  // store new todo in memory with data from request body
+  var newTodo = new Todo(req.body.todo);
+
+  // find list in db by id and add new todo
+  List.findOne({_id: listId}, function (err, foundList) {
+    foundList.todos.push(newTodo);
+    foundList.save(function (err, savedList) {
+      res.json(newTodo);
+    });
+  });
+});
+```
+
+### Route to Update Embedded Data
+
+```js
+// update todo embedded in list
+app.put('/api/lists/:listId/todos/:id', function (req, res) {
+  // set the value of the list and todo ids
+  var listId = req.params.listId;
+  var todoId = req.params.id;
+
+  // find list in db by id
+  List.findOne({_id: listId}, function (err, foundList) {
+    // find todo embedded in list
+    var foundTodo = foundList.todos.id(todoId);
+    // update todo text and completed with data from request body
+    foundTodo.text = req.body.todo.text;
+    foundTodo.completed = req.body.todo.completed;
+    foundList.save(function (err, savedList) {
+      res.json(foundTodo);
+    });
+  });
+});
+```
+
+### Route to Delete Embedded Data
+
+```js
+// delete todo embedded in list
+app.delete('/api/lists/:listId/todos/:id', function (req, res) {
+  // set the value of the list and todo ids
+  var listId = req.params.listId;
+  var todoId = req.params.id;
+
+  // find list in db by id
+  List.findOne({_id: listId}, function (err, foundList) {
+    // find todo embedded in list
+    var foundTodo = foundList.todos.id(todoId);
+    // remove todo
+    foundTodo.remove();
+    foundList.save(function (err, savedList) {
+      res.json(foundTodo);
+    });
+  });
+});
+```
+
+## Challenges
+
+**Setting up an Embedded Model**
+
+1. Write a file called `post.js` that has a post model with an attribute `comments` that contains an embedded comment schema.
+
+**Writing User/Tweets Routes**
+
+The following challenges are psuedo code. Write a method for example.
+
```

Comment:
@ben-manning Sorry for the late comment on this one, but the instructions about pseudo code are confusing here. There are no solutions for anything related to a monster - are these just examples? If so, maybe move them above the word "challenges."